### PR TITLE
gh-102598: Remove obsolete optimization from `FORMAT_VALUE` opcode

### PR DESCRIPTION
**Pull request by sobolevn**
Since `PyObject_FormatValue` already has this check, there's no need to duplicate it here.

I want to apply `:refleak:` label to be sure that we have the same refcount in the optimized case (it looks like it, just to be sure).

Skipping news, because there's no user-facing change.

* Issue: gh-102598

### CODE REVIEW

**Comment by sobolevn** [comment]
@brandtbucher GitHub suggested you :)

**Review by hauntsaninja — APPROVED** [review]
This looks reasonable!

Commit for the "I plan to" comment: https://github.com/python/cpython/commit/a78c795
Commit for the check in PyObject_Format: https://github.com/python/cpython/commit/ea525a2

I guess there's an extra incref decref

**Comment by sobolevn** [comment]
@gvanrossum is a code-owner of these file, maybe you would have time to take a look? :)

File: Python/bytecodes.c [hunk 1]
```diff
@@ -3026,20 +3026,11 @@ dummy_func(
...
-            if (PyUnicode_CheckExact(value) && fmt_spec == NULL) {
-                /* Do nothing, just transfer ownership to result. */
-                result = value;
-            } else {
-                /* Actually call format(). */
-                result = PyObject_Format(value, fmt_spec);
-                DECREF_INPUTS();
-                ERROR_IF(result == NULL, error);
-            }
+            /* Actually call format(). */
```
Comment (gvanrossum) [inline Python/bytecodes.c]: This comment now seems redundant.
```suggestion
```

**Review by gvanrossum — COMMENTED** [review]
LG, one nit.

**Pull request merged**